myriad-scheduler/src/main/java/com/ebay/myriad/state/utils/StoreContext.java

```diff
@@ -0,0 +1,273 @@
+  }
+
+  /**
+   * Accept list of ByteBuffers and re-create the SchedulerState maps.
+   * @param framwrorkId
+   * @param taskIds
+   * @param taskNodes
+   * @param pendingTasks
+   * @param stagingTasks
+   * @param activeTasks
+   * @param lostTasks
+   * @param killableTasks
+   */
+  public StoreContext(ByteBuffer frameworkId,
+    List<ByteBuffer> taskIds, List<ByteBuffer> taskNodes,
+    List<ByteBuffer> pendingTasks, List<ByteBuffer> stagingTasks,
+    List<ByteBuffer> activeTasks, List<ByteBuffer> lostTasks,
+    List<ByteBuffer> killableTasks) {
+    this.frameworkId = frameworkId;
+    this.taskIds = taskIds;
+    this.taskNodes = taskNodes;
+    this.pendingTasks = pendingTasks;
+    this.stagingTasks = stagingTasks;
+    this.activeTasks = activeTasks;
+    this.lostTasks = lostTasks;
+    this.killableTasks = killableTasks;
+  }
+
+  /**
+   * Use this to gather bytes to push to the state store
+   * @return byte stream of the state store context.
+   * @throws IOException
+   */
+  public ByteArrayOutputStream toSerializedContext() throws IOException {
+    ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+    ByteBufferSupport.addByteBuffer(frameworkId, bytes);
+    ByteBufferSupport.addByteBuffers(taskIds, bytes);
+    ByteBufferSupport.addByteBuffers(taskNodes, bytes);
+    ByteBufferSupport.addByteBuffers(pendingTasks, bytes);
+    ByteBufferSupport.addByteBuffers(stagingTasks, bytes);
+    ByteBufferSupport.addByteBuffers(activeTasks, bytes);
+    ByteBufferSupport.addByteBuffers(lostTasks, bytes);
+    ByteBufferSupport.addByteBuffers(killableTasks, bytes);
+    return bytes;
+  }
+
+  /**
+   * When the bytes come back from the store, use this method to create a new context.
+   *
+   * @param bytes from state store
+   * @return initialized StoreContext to use to initialize a SchedulerState
+   */
+  @SuppressWarnings("unchecked")
+  public static StoreContext fromSerializedBytes(byte bytes[]) {
+    StoreContext ctx;
+    if (bytes != null && bytes.length > 0){
+      ByteBuffer bb = ByteBufferSupport.fillBuffer(bytes);
+      ByteBuffer frameworkId = ByteBufferSupport.createBuffer(bb);
+      List<ByteBuffer> taskIds = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> taskNodes = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> pendingTasks = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> stagingTasks = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> activeTasks = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> lostTasks = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      List<ByteBuffer> killableTasks = ByteBufferSupport.createBufferList(bb, bb.getInt());
+      ctx = new StoreContext(frameworkId, taskIds, taskNodes, pendingTasks, stagingTasks, activeTasks,
+        lostTasks, killableTasks);
+    } else {
+      ctx = new StoreContext();
```

Comment:
SchedulerState.loadStateStore(). However I am changing the MyriadStateStore interface to accept and return the StoreContext object, so this will now move to MyriadFileSystemRMStateStore.loadMyriadState